fluss-server/src/main/java/org/apache/fluss/server/zk/data/lake/LakeTableJsonSerde.java

```diff
@@ -61,28 +61,35 @@ public class LakeTableJsonSerde implements JsonSerializer<LakeTable>, JsonDeseri
 
     @Override
     public void serialize(LakeTable lakeTable, JsonGenerator generator) throws IOException {
-        generator.writeStartObject();
-        generator.writeNumberField(VERSION_KEY, CURRENT_VERSION);
-
-        generator.writeArrayFieldStart(LAKE_SNAPSHOTS);
-        for (LakeTable.LakeSnapshotMetadata lakeSnapshotMetadata :
-                checkNotNull(lakeTable.getLakeSnapshotMetadatas())) {
+        // if lake table snapshot metadata is null, it must be version 1
+        if (lakeTable.getLakeSnapshotMetadatas() == null) {
+            // Version 1: ZK node contains full snapshot data, use LakeTableSnapshotJsonSerde
+            LakeTableSnapshotJsonSerde.INSTANCE.serialize(
+                    lakeTable.getLatestTableSnapshot(), generator);
+        } else {
             generator.writeStartObject();
+            generator.writeNumberField(VERSION_KEY, CURRENT_VERSION);
+
+            generator.writeArrayFieldStart(LAKE_SNAPSHOTS);
+            for (LakeTable.LakeSnapshotMetadata lakeSnapshotMetadata :
```

Comment:
It would be better to extract separate `serializeV1()` and `serializeV2()` methods to improve maintainability and readability. The same applies to the `deserialize`.